pipeline/transformer/transformer_common/src/main/scala/weco/catalogue/transformer/TransformerWorker.scala

```diff
@@ -104,10 +148,16 @@ trait TransformerWorker[Payload <: SourcePayload, SourceData, SenderDest]
               case TransformerError(_, sourceData, key) =>
                 error(s"$name: TransformerError on $sourceData with $key")
             }
-            Future.failed(err)
-          case Right((work, key)) =>
+
+            throw err
+
+          case Right((None, key)) =>
+            info(s"$name: from $key transformed work; already in pipeline")
```

Comment:
nitpicking, but why do we return the `key` all the way to here only to log and discard it? can we not log in the `compareToStored` method and avoid returning? the return type of `compareToStored` is a bit confusing to me

Reply:
Good shout.

Reply:
Fixed.